clang/labs/lab1/polozov_vladislav/WarningForDeprecated.cpp

```diff
@@ -0,0 +1,73 @@
+#include "clang/AST/ASTConsumer.h"
+#include "clang/AST/RecursiveASTVisitor.h"
+#include "clang/Frontend/CompilerInstance.h"
+#include "clang/Frontend/FrontendPluginRegistry.h"
+#include "llvm/Support/raw_ostream.h"
+
+using namespace clang;
+
+class WarningDeprecatedConsumer : public ASTConsumer {
+  CompilerInstance &Instance;
+
+public:
+  WarningDeprecatedConsumer(CompilerInstance &Instance) : Instance(Instance) {}
+
+  void HandleTranslationUnit(ASTContext &context) override {
+
+    struct Visitor : public RecursiveASTVisitor<Visitor> {
+      ASTContext *Context;
+      Visitor(ASTContext *_Context) : Context(_Context) {}
+      bool VisitFunctionDecl(FunctionDecl *FD) {
+        std::string want_find = "deprecated";
+        bool find = false;
+        std::string name = FD->getNameInfo().getAsString();
+        for (int i = 0; i + want_find.size() - 1 < name.size(); i++) {
+          bool ok = true;
+          for (int j = 0; j < want_find.size(); j++) {
+            if (want_find[j] != name[i + j]) {
+              ok = false;
+              break;
+            }
+          }
+          if (ok) {
+            find = true;
+            break;
+          }
+        }
```

Comment:
Could you please use `std::string` comparison methods for checking strings equity?
Also could you please add option for case insensitive check of `deprecated`?